src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala

```diff
@@ -597,7 +597,7 @@ final case class ClientApiServiceImpl(
         if (roles.contains(SECURITY_ROLE)) List(getRelationship(requesterUuid, userUuid)).sequence
         else parseArrayParameters(relationshipIds).traverse(_.toFutureUUID)
```

Comment:
This comment is not strictly related to this PR only.
Here we are using the cats functions on Future which is not recommended.
How do we handle them in our code base?
Should we leave them as they are, waiting to use a different stack, or should we 
change them?
@galales @TonioGela

Reply:
Well, in this specific case I'll change the first line to:

`if (roles.contains(SECURITY_ROLE)) getRelationship(requesterUuid, userUuid).map(List)`

while the second effectively traverses a List and executes code that is synchronous so I won't worry that much: 

`else parseArrayParameters(relationshipIds).traverse(_.toFutureUUID)`

As per the general picture, I don't know, the future strictly depends on which stack we'll use, and cats will either be the core of it or be completely useless. For the moment I'll say let's keep at least an eye on NOT traversing a list to make API calls to other services.

Reply:
Antonio,
if (roles.contains(SECURITY_ROLE)) getRelationship(requesterUuid, userUuid).map(List)

does not compile.

Where am I wrong?

Reply:
It absolutely may not compile, the idea is to take the result of `getRelationship(requesterUuid, userUuid)` that may be a `Future[X]` and use map to put that `X` in a `List[X]` to get a `Future[List[X]]`

Probably `.map(List)` should be written as `.map(List.apply)` or `.map(List(_)` or `.map(List[X](_))` or `.map(_ :: Nil)`. The idea is to find a way to call the `List` constructor :D

Reply:
Just done